tests/signal_manager_test.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+include './src/signal_manager.sh'
+include './tests/utils'
+
+function test_default_handler()
+{
+  local -r original_dir="$PWD"
+  local -r expected=$'\n'"Oh no! An interruption! See ya..."
+
+  cd "$SHUNIT_TMPDIR"
+
+  signal_manager 'kill -s SIGTERM $!' SIGUSR1
+  {
+    signal_manager default_interrupt_handler SIGINT SIGTERM
+    sleep infinity &
+    kill -s SIGUSR1 $$ # Send SIGUSR1 to parent process
+    wait $! # wait for SIGTERM
```

Comment:
just for curiosity, should you double-quote these variables? What shellcheck says?